src/check.py

```diff
@@ -25,6 +29,50 @@ class Problem(NamedTuple):
     description: str
 
 
+format_path = "/".join
+
+
+class TrackPathsWithDiff:
+    """Keeps track of paths that have any differences.
+
+    Attrs:
+        base_local_diffs: The paths that have a difference between the base and local content.
+        base_server_diffs: The paths that have a difference between the local and server content.
+    """
+
+    def __init__(self) -> None:
+        """Construct."""
+        self._base_local_diffs: list[str] = []
+        self._base_server_diffs: list[str] = []
+
+    @property
+    def base_local_diffs(self) -> Iterator[str]:
+        """Get the paths with base and local diffs."""
+        return iter(self._base_local_diffs)
+
+    @property
+    def base_server_diffs(self) -> Iterator[str]:
+        """Get the paths with local and server diffs."""
+        return iter(self._base_server_diffs)
+
+    def process(self, action: UpdateAction) -> None:
+        """Record differences for a given action.
+
+        Args:
+            action: The update action to process.
+        """
+        content_change = action.content_change
+
+        if content_change is None:
+            return
+
+        if content_change.base != content_change.local:
+            self._base_local_diffs.append(format_path(action.path))
+
+        if content_change.base != content_change.server:
+            self._base_server_diffs.append(format_path(action.path))
```

Comment:
I'm really not a big fan of mutable properties. Is this really necessary? Can't we use a classmethod to construct an immutable object, and operations to eventually add immutable objects?

Reply:
Changed to take an iterable actions in the init instead of accumulating